work/v1/types.go

```diff
@@ -225,3 +225,68 @@ type ManifestWorkList struct {
 	// Items is a list of manifestworks.
 	Items []ManifestWork `json:"items"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:subresource:status
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// AppliedManifestWork represents an applied manifestwork on managed cluster. It is placed
+// on managed cluster. An AppliedManifestWork links to a manifestwork on a hub recording resources
+// deployed in the managed cluster.
+// When the agent is removed from managed cluster, cluster-admin on managed cluster
+// can delete appliedmanifestwork to remove resources deployed by the agent.
+type AppliedManifestWork struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired configuration of AppliedManifestWork
+	Spec AppliedManifestWorkSpec `json:"spec,omitempty"`
+
+	// Status represents the current status of AppliedManifestWork
+	// +optional
+	Status AppliedManifestWorkStatus `json:"status,omitempty"`
+}
+
+// AppliedManifestWorkSpec represents the desired configuration of AppliedManifestWork
+type AppliedManifestWorkSpec struct {
+	// HubID represents the ID of the hub that the agent fetches ManifestWork. It could
```

Comment:
I think we want to be more specific about what this is - where does this correspond to a type defined elsewhere in this repo?

Reply:
> I think we want to be more specific about what this is - where does this correspond to a type defined elsewhere in this repo?

Perhaps we can avoid it by declaring it a hub hash created by hashing the *first* hub kube apiserver URL.